### PR TITLE
Close socket

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

**Comment by MrCreosote** [comment]
Would a `with` work better here?

**Comment by Tianhao-Gu** [comment]
> Would a `with` work better here?

**Comment by Tianhao-Gu** [comment]
@MrCreosote using `with requests`. There are two requests in try-except block. I think using `response.close()` is more clean.

**Comment by MrCreosote** [comment]
`with` ensures that whatever happens in the block, the socket is closed. If you don't use `with` and there's an error in the block prior to the close call you'll leak sockets

**Comment by Tianhao-Gu** [comment]
@MrCreosote Sure. Updated all `requests` using `with`.

File: lib/DataFileUtil/DataFileUtilImpl.py [hunk 1]
```diff
@@ -377,13 +377,7 @@ def _retrieve_filepath(self, file_url, cookies=None):
 
         """
 
-        try:
```
Comment (MrCreosote) [inline lib/DataFileUtil/DataFileUtilImpl.py]: You can mix try and with statements: https://stackoverflow.com/questions/713794/catching-an-exception-while-using-a-python-with-statement

IMO we shouldn't change the API unless we have to.

Comment (Tianhao-Gu) [reply lib/DataFileUtil/DataFileUtilImpl.py]: 👍 copy that.

**Comment by MrCreosote** [comment]
LGTM

**Pull request merged**